src/Microsoft.Restier.WebApi/RestierController.cs

```diff
@@ -55,6 +55,7 @@ public ApiBase Api
                 if (this.api == null)
                 {
                     this.api = this.Request.GetApiFactory().Invoke();
+                    this.Request.SetApiInstance(this.api);
```

Comment:
Yeah, I move the setting into controller select and did some test with post, it is called before deserializer.